fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidHavingBuilder.java

```diff
@@ -68,15 +67,34 @@ public static Having buildMetricHaving(LogicalMetric metric, Set<ApiHaving> havi
     /**
      * Create a list of NumericHavings for the values specified and OR them together.
      *
-     * @param metric  The metric that the operation applied to.
-     * @param having  The ApiHaving object
-     *
+     * @param metric The metric that the operation applied to.
+     * @param having The ApiHaving object
      * @return A single having representing the API Filter
      */
     public static Having buildHaving(LogicalMetric metric, ApiHaving having) {
         LOG.trace("Building having using metric: {} and API Having: {}", metric, having);
 
         HavingOperation operation = having.getOperation();
+
+        Set<Double> values = having.getValues();
```

Comment:
Let's actually get a little more firm in parsing the values list.  Right now your code will work if between lists three values, which is nice, except that if someone asks the question with two values, they've probably made a bad mistake.  Better to validate both count and order on a having-style operation and error if they are wrong.